src/ResourceManager/Network/Commands.Network/VirtualNetworkGateway/GenerateAzureVpnClientPackage.cs

```diff
@@ -65,10 +78,164 @@ public override void ExecuteCmdlet()
+        {
+            #region 1. Send Async request to generate vpn client package
+
+            // 1. Send Async request to generate vpn client package          
+            string baseUrl = this.NetworkClient.NetworkManagementClient.BaseUri.ToString();
+            string apiVersion = this.NetworkClient.NetworkManagementClient.ApiVersion;
+
+            if (resourceGroupName == null)
+            {
+                throw new ValidationException(ValidationRules.CannotBeNull, "resourceGroupName");
+            }
+            if (virtualNetworkGatewayName == null)
+            {
+                throw new ValidationException(ValidationRules.CannotBeNull, "virtualNetworkGatewayName");
+            }
+            if (parameters == null)
+            {
+                throw new ValidationException(ValidationRules.CannotBeNull, "parameters");
+            }
+
+            // Construct URL
+            var url = new Uri(new Uri(baseUrl + (baseUrl.EndsWith("/") ? "" : "/")), "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualnetworkgateways/{virtualNetworkGatewayName}/generatevpnclientpackage").ToString();
+            url = url.Replace("{resourceGroupName}", Uri.EscapeDataString(resourceGroupName));
+            url = url.Replace("{virtualNetworkGatewayName}", Uri.EscapeDataString(virtualNetworkGatewayName));
+            url = url.Replace("{subscriptionId}", Uri.EscapeDataString(this.NetworkClient.NetworkManagementClient.SubscriptionId));
+            url += "?" + string.Join("&", string.Format("api-version={0}", Uri.EscapeDataString(apiVersion)));
+
+            // Create HTTP transport objects
+            HttpRequestMessage httpRequest = new HttpRequestMessage();
+            httpRequest.Method = new HttpMethod("POST");
+            httpRequest.RequestUri = new Uri(url);
+            // Set Headers
+            httpRequest.Headers.TryAddWithoutValidation("x-ms-client-request-id", Guid.NewGuid().ToString());
+
+            // Serialize Request
+            string requestContent = JsonConvert.SerializeObject(parameters, this.NetworkClient.NetworkManagementClient.SerializationSettings);
+            httpRequest.Content = new StringContent(requestContent, Encoding.UTF8);
+            httpRequest.Content.Headers.ContentType = MediaTypeHeaderValue.Parse("application/json; charset=utf-8");
+
+            // Set Credentials
+            if (this.NetworkClient.NetworkManagementClient.Credentials != null)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                await this.NetworkClient.NetworkManagementClient.Credentials.ProcessHttpRequestAsync(httpRequest, cancellationToken).ConfigureAwait(false);
+            }
+            // Send Request
+            cancellationToken.ThrowIfCancellationRequested();
+            HttpClient httpClient = new HttpClient();
```

Comment:
why not use the http client from NetworkManagementClient here? (i.e. this.NetworkClient.NetworkManagementClient.HttpClient) This way you gate all the custom configuration that PowerShell does of that client